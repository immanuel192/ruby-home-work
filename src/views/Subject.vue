<script lang="ts">
import {
  computed, defineComponent,
} from '@vue/composition-api';
import AppstorePage from '@/components/AppstorePage.vue';
import AppstorePageTitle from '@/components/AppstorePageTitle.vue';
import AppstoreCard from '@/components/AppstoreCard.vue';

import { reactiveDate } from '@/reactive';
import { injectStore } from '@/share';

export default defineComponent({
  name: 'today',
  components: {
    AppstorePage,
    AppstorePageTitle,
    AppstoreCard,
  },
  setup () {
    const store = injectStore();
    const subjects = computed(() => store.getters.groupedSubjects);
    const firstSubject = subjects.value[0].key;
    const { currentDateTitle: todayDateTitle } = reactiveDate();

    return {
      todayDateTitle,
      subjects,
      firstSubject,
    };
  },
});
</script>

<template>
  <appstore-page noCollapsedNavbar>
    <template v-for="groupedSubject in subjects">
      <appstore-page-title :key="`${groupedSubject.key}-title`"
        :title="groupedSubject.key"
        :heading="groupedSubject.key === firstSubject ? todayDateTitle : ''"
        :accountLink="groupedSubject.key === firstSubject " />
      <div class="appstore-card-grid" :key="`${groupedSubject.key}-main`">
        <appstore-card
          v-for="subject in groupedSubject.items"
          :key="subject.id"
          :subjectId="subject.id"
          v-bind="subject.styleConfig" >
        </appstore-card>
      </div>
    </template>
  </appstore-page>
</template>
