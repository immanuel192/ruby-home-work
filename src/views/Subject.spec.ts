import { ref } from '@vue/composition-api';
import Subject from '@/views/Subject.vue';
import {
  shallowMount, suiteName,
  createMockStore, IStoreMock,
} from '@tests/util';
import * as reactives from '@/reactive';

describe(suiteName(__filename), () => {
  let store: IStoreMock;

  beforeAll(() => {
    store = createMockStore();
  });

  it('should matched snapshot', () => {
    store.getters.groupedSubjects = [
      {
        key: 'subject1',
        items: [
          {
            enable: true,
            group: 'group1',
            id: 'id1',
            name: 'name1',
          },
          {
            enable: true,
            group: 'group2',
            id: 'id2',
            name: 'name2',
          },
        ],
      },
      {
        key: 'subject2',
        items: [
          {
            enable: true,
            group: 'group3',
            id: 'id3',
            name: 'name3',
          },
        ],
      },
    ];

    jest.spyOn(reactives, 'reactiveDate').mockReturnValue({
      currentDateTitle: ref('Today date'),
    });
    const wrapper = shallowMount(Subject, {
      provide: { store },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
