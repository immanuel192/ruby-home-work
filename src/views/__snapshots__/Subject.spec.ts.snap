// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ruby-home-work#src#views#Subject.spec.ts should matched snapshot 1`] = `
<appstore-page-stub title="" navbarheading="" nocollapsednavbar="true">
  <appstore-page-title-stub title="subject1" heading="Today date" accountlink="true"></appstore-page-title-stub>
  <div class="appstore-card-grid">
    <appstore-card-stub imagealt="" title="" titlecolor="#000" subtitle="" subtitlecolor="#000" titleposition="top" appcolor="#fff" closebuttoncolor="#fff" apps="" subjectid="id1"></appstore-card-stub>
    <appstore-card-stub imagealt="" title="" titlecolor="#000" subtitle="" subtitlecolor="#000" titleposition="top" appcolor="#fff" closebuttoncolor="#fff" apps="" subjectid="id2"></appstore-card-stub>
  </div>
  <appstore-page-title-stub title="subject2" heading=""></appstore-page-title-stub>
  <div class="appstore-card-grid">
    <appstore-card-stub imagealt="" title="" titlecolor="#000" subtitle="" subtitlecolor="#000" titleposition="top" appcolor="#fff" closebuttoncolor="#fff" apps="" subjectid="id3"></appstore-card-stub>
  </div>
</appstore-page-stub>
`;
